src/zcl_abapgit_objects.clas.abap

```diff
@@ -20,6 +20,29 @@ CLASS zcl_abapgit_objects DEFINITION
         files TYPE zif_abapgit_definitions=>ty_files_tt,
         item  TYPE zif_abapgit_definitions=>ty_item,
       END OF ty_serialization .
+    TYPES:
+      ty_deserialization_step TYPE string.
+    TYPES:
+      ty_deserialization_step_tt TYPE STANDARD TABLE OF ty_deserialization_step
+                                            WITH DEFAULT KEY .
+    TYPES:
+      BEGIN OF ty_step_data,
+        id           TYPE string,
+        order        TYPE i,
+        descr        TYPE string,
+        is_ddic      TYPE abap_bool,
+        syntax_check TYPE abap_bool,
+        objects      TYPE ty_deserialization_tt,
+      END OF ty_step_data.
+    TYPES:
+      ty_step_data_tt TYPE STANDARD TABLE OF ty_step_data
```

Comment:
can these 2 types `ty_step_data*` be moved to private?

Reply:
the types are used in the interface zif_abapgit_object.
I suggest to later move both types to zif_abapgit_definitions is the change works well.

Reply:
Then I suggest moving it to the `zif_abapgit_object` interface, having interfaces refer class types breaks the reason for having interfaces

Reply:
moved types to the IF